7.6 RC3: DependencyAdder.invoke cannot be called with the arguments supplied

### Expected Behavior
Precompiled script plugins [configuring a project dependency for an additional test suite](https://docs.gradle.org/current/userguide/jvm_test_suite_plugin.html#declare_an_additional_test_suite) via the JVM Test Site Plugin continue to compile after upgrading to Gradle 7.6 RC3.

### Current Behavior
The script fails to compile:

```stacktrace
e: C:\Dev\Repos\SDKotlin\sd-kotlin-talks\build-logic\src\main\kotlin\org.sdkotlin.build.test.integration-test-suite.gradle.kts: (20, 5): None of the following functions can be called with the arguments supplied: 
public operator fun DependencyAdder.invoke(dependencyNotation: CharSequence): Unit defined in org.gradle.kotlin.dsl
public operator fun DependencyAdder.invoke(dependency: Dependency): Unit defined in org.gradle.kotlin.dsl
public operator fun DependencyAdder.invoke(files: FileCollection): Unit defined in org.gradle.kotlin.dsl
public operator fun DependencyAdder.invoke(dependency: Provider<out Dependency>): Unit defined in org.gradle.kotlin.dsl
public operator fun DependencyAdder.invoke(externalModule: ProviderConvertible<out MinimalExternalModuleDependency>): Unit defined in org.gradle.kotlin.dsl

FAILURE: Build failed with an exception.
```

### Context
I'm unable to upgrade to Gradle 7.6.

### Steps to Reproduce 
Reproducer: https://github.com/sdkotlin/sd-kotlin-talks/tree/a54afe81013c1806550ff23478efe48973033deb

Line in error: https://github.com/sdkotlin/sd-kotlin-talks/blob/a54afe81013c1806550ff23478efe48973033deb/build-logic/src/main/kotlin/org.sdkotlin.build.test.integration-test-suite.gradle.kts#L20

### Your Environment
Gradle 7.6-rc3
Kotlin 1.7.21
Java 11.0.17

--- a/subprojects/docs/src/docs/userguide/jvm/jvm_test_suite_plugin.adoc
+++ b/subprojects/docs/src/docs/userguide/jvm/jvm_test_suite_plugin.adoc
@@ -95,6 +95,7 @@ Common values are available as constants in link:{javadocPath}/org/gradle/api/at
 
 Here are several examples to illustrate the configurability of test suites.
 
+[[sec:declare_an_additional_test_suite]]
 == Declare an additional test suite
 ====
 include::sample[dir="snippets/testing/test-suite-plugin/groovy",files="build.gradle[tags=configure-testing-extension]"]

--- a/subprojects/docs/src/docs/userguide/jvm/jvm_test_suite_plugin.adoc
+++ b/subprojects/docs/src/docs/userguide/jvm/jvm_test_suite_plugin.adoc
@@ -222,6 +222,7 @@ include::sample[dir="snippets/testing/test-suite-multi-configure-each-extracted/
 <2> Apply the closure to a test suite, using the default (`test`) test suite
 <3> Alternate means of applying a configuration closure to a test suite outside of its declaration, using the `integrationTest` test suite
 
+[[sec:differences_with_top_level_dependencies]]
 == Differences between the test suite `dependencies` and the top-level `dependencies` blocks
 
 Gradle 7.6 changed the API of the test suite's `dependencies` block to be more strongly-typed.

--- a/subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc
+++ b/subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc
@@ -117,9 +117,9 @@ PMD has been updated to https://pmd.github.io/pmd-6.48.0/pmd_release_notes.html[
 When configuring an executable explicitly for link:{groovyDslPath}/org.gradle.api.tasks.compile.ForkOptions.html#org.gradle.api.tasks.compile.ForkOptions:executable[`JavaCompile`] or link:{groovyDslPath}/org.gradle.api.tasks.testing.Test.html#org.gradle.api.tasks.testing.Test:executable[`Test`] tasks, Gradle will now emit an error if this executable does not exist.
 In the past, the task would be executed with the default toolchain or JVM running the build.
 
-==== Changes to dependency declarateions in Test Suites
+==== Changes to dependency declarations in Test Suites
 
-As part of the ongoing effort to improve Test Suites, dependency declarations using its nested `dependencies` block are <<jvm_test_suite_plugin.adoc#sec:differences_with_top_level_dependencies, now strongly typed>>.
+As part of the ongoing effort to evolve Test Suites, dependency declarations in the Test Suites `dependencies` block are <<jvm_test_suite_plugin.adoc#sec:differences_with_top_level_dependencies, now strongly typed>>.
 This will help make this incubating API more discoverable and easier to use in an IDE.
 
 In some cases, this requires syntax changes.

--- a/subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc
+++ b/subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc
@@ -117,6 +117,56 @@ PMD has been updated to https://pmd.github.io/pmd-6.48.0/pmd_release_notes.html[
 When configuring an executable explicitly for link:{groovyDslPath}/org.gradle.api.tasks.compile.ForkOptions.html#org.gradle.api.tasks.compile.ForkOptions:executable[`JavaCompile`] or link:{groovyDslPath}/org.gradle.api.tasks.testing.Test.html#org.gradle.api.tasks.testing.Test:executable[`Test`] tasks, Gradle will now emit an error if this executable does not exist.
 In the past, the task would be executed with the default toolchain or JVM running the build.
 
+==== Strongly typed dependency declarations for Test Suites
+
+The `dependencies` block used to configure Test Suites and located at `testing.suites.<MY_SUITE_NAME>.dependencies` continues to diverge from the top-level `dependencies` block.
+This is part of an ongoing effort to improve this `@Incubating` API to make it more discoverable and strongly typed.
+This change should have the benefit of making the `testing` DSL easier to work with in your IDE.
+
+On consequence of this is that buildsscripts which previously added project dependencies to a suite using:
+
+```kotlin
+testing {
+  suites {
+    register<JvmTestSuite>("integrationTest") {
+      dependencies {
+        implementation(project)
+      }
+    }
+  }
+}
+```
+
+will now fail to compile, with a message like:
+
+```
+None of the following functions can be called with the arguments supplied:
+public operator fun DependencyAdder.invoke(dependencyNotation: CharSequence): Unit defined in org.gradle.kotlin.dsl
+public operator fun DependencyAdder.invoke(dependency: Dependency): Unit defined in org.gradle.kotlin.dsl
+public operator fun DependencyAdder.invoke(files: FileCollection): Unit defined in org.gradle.kotlin.dsl
+public operator fun DependencyAdder.invoke(dependency: Provider<out Dependency>): Unit defined in org.gradle.kotlin.dsl
+public operator fun DependencyAdder.invoke(externalModule: ProviderConvertible<out MinimalExternalModuleDependency>): Unit defined in org.gradle.kotlin.dsl
+```
+
+To fix this, replace the reference to `project` with a call to `project()`:
+
+```kotlin
+testing {
+  suites {
+    register<JvmTestSuite>("integrationTest") {
+      dependencies {
+        implementation(project())
+      }
+    }
+  }
+}
+```
+
+There may be other breaking changes related to dependency declarations in this block.
+For more information, see the updated <<jvm_test_suite_plugin.adoc#sec:declare_an_additional_test_suite, declare an additional test suite>> example in the JVM Test Suite Plugin section of the user guide, and the link:{groovyDslPath}/org.gradle.api.artifacts.dsl.DependencyAdder.html[`DependencyAdder`] page in the DSL reference.
+
+```kotlin
+
 === Deprecations
 
 [[invalid_toolchain_specification_deprecation]]

--- a/subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc
+++ b/subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc
@@ -117,13 +117,11 @@ PMD has been updated to https://pmd.github.io/pmd-6.48.0/pmd_release_notes.html[
 When configuring an executable explicitly for link:{groovyDslPath}/org.gradle.api.tasks.compile.ForkOptions.html#org.gradle.api.tasks.compile.ForkOptions:executable[`JavaCompile`] or link:{groovyDslPath}/org.gradle.api.tasks.testing.Test.html#org.gradle.api.tasks.testing.Test:executable[`Test`] tasks, Gradle will now emit an error if this executable does not exist.
 In the past, the task would be executed with the default toolchain or JVM running the build.
 
-==== Strongly typed dependency declarations for Test Suites
+==== Added strongly typed dependency declarations to test suites
 
-The `dependencies` block used to configure Test Suites and located at `testing.suites.<MY_SUITE_NAME>.dependencies` continues to diverge from the top-level `dependencies` block.
-This is part of an ongoing effort to improve this `@Incubating` API to make it more discoverable and strongly typed.
-This change should have the benefit of making the `testing` DSL easier to work with in your IDE.
-
-On consequence of this is that buildsscripts which previously added project dependencies to a suite using:
+As part of the ongoing effort to improve the incubating Test Suites `dependencies` block, dependency declarations are now strongly typed.
+In some case, this requires syntax changes.
+For example, build scripts that previously added test suite dependencies with the following syntax:
 
 ```kotlin
 testing {

--- a/subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc
+++ b/subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc
@@ -162,7 +162,7 @@ testing {
 
 There may be other breaking changes related to dependency declarations in this block.
 
-For more information, see the updated <<jvm_test_suite_plugin.adoc#sec:declare_an_additional_test_suite, declare an additional test suite>> example in the JVM Test Suite Plugin section of the user guide, and the link:{groovyDslPath}/org.gradle.api.artifacts.dsl.DependencyAdder.html[`DependencyAdder`] page in the DSL reference.
+For more information, see the updated <<jvm_test_suite_plugin.adoc#sec:declare_an_additional_test_suite, declare an additional test suite>> example in the JVM Test Suite Plugin section of the user guide and the link:{groovyDslPath}/org.gradle.api.artifacts.dsl.DependencyAdder.html[`DependencyAdder`] page in the DSL reference.
 
 === Deprecations
 

--- a/subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc
+++ b/subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc
@@ -166,7 +166,7 @@ Other syntax effected by this change includes:
 
 - You cannot use `Provider<String>` as a dependency declaration.
 - You cannot use a `Map` as a dependency declaration for Kotlin or Java.
-- You cannot use a bundle as a dependency declaration directly (`implementation(libs.bundles.testing)`), instead you have to use `implementation.bundle(libs.bundles.testing)`.
+- You cannot use a bundle as a dependency declaration directly (`implementation(libs.bundles.testing)`). Use `implementation.bundle(libs.bundles.testing)` instead.
 
 For more information, see the updated <<jvm_test_suite_plugin.adoc#sec:declare_an_additional_test_suite, declare an additional test suite>> example in the JVM Test Suite Plugin section of the user guide and the link:{groovyDslPath}/org.gradle.api.artifacts.dsl.DependencyAdder.html[`DependencyAdder`] page in the DSL reference.
 

--- a/subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc
+++ b/subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc
@@ -122,7 +122,7 @@ In the past, the task would be executed with the default toolchain or JVM runnin
 As part of the ongoing effort to improve Test Suites, dependency declarations using its nested `dependencies` block are <<jvm_test_suite_plugin.adoc#sec:differences_with_top_level_dependencies, now strongly typed>>.
 This will help make this incubating API more discoverable and easier to use in an IDE.
 
-In some case, this requires syntax changes.
+In some cases, this requires syntax changes.
 For example, build scripts that previously added Test Suite dependencies with the following syntax:
 
 ```kotlin

--- a/subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc
+++ b/subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc
@@ -117,11 +117,13 @@ PMD has been updated to https://pmd.github.io/pmd-6.48.0/pmd_release_notes.html[
 When configuring an executable explicitly for link:{groovyDslPath}/org.gradle.api.tasks.compile.ForkOptions.html#org.gradle.api.tasks.compile.ForkOptions:executable[`JavaCompile`] or link:{groovyDslPath}/org.gradle.api.tasks.testing.Test.html#org.gradle.api.tasks.testing.Test:executable[`Test`] tasks, Gradle will now emit an error if this executable does not exist.
 In the past, the task would be executed with the default toolchain or JVM running the build.
 
-==== Added strongly typed dependency declarations to test suites
+==== Changes to dependency declarateions in Test Suites
 
-As part of the ongoing effort to improve the incubating Test Suites `dependencies` block, dependency declarations are now strongly typed.
+As part of the ongoing effort to improve Test Suites, dependency declarations using its nested `dependencies` block are <<jvm_test_suite_plugin.adoc#sec:differences_with_top_level_dependencies, now strongly typed>>.
+This will help make this incubating API more discoverable and easier to use in an IDE.
+
 In some case, this requires syntax changes.
-For example, build scripts that previously added test suite dependencies with the following syntax:
+For example, build scripts that previously added Test Suite dependencies with the following syntax:
 
 ```kotlin
 testing {
@@ -160,7 +162,11 @@ testing {
 }
 ```
 
-There may be other breaking changes related to dependency declarations in this block.
+There are other related changes, such as:
+
+- You cannot use `Provider<String>` as a dependency declaration.
+- You cannot use a `Map` as a dependency declaration for Kotlin or Java.
+- You cannot use a bundle as a dependency declaration directly (`implementation(libs.bundles.testing)`), instead you have to use `implementation.bundle(libs.bundles.testing)`.
 
 For more information, see the updated <<jvm_test_suite_plugin.adoc#sec:declare_an_additional_test_suite, declare an additional test suite>> example in the JVM Test Suite Plugin section of the user guide and the link:{groovyDslPath}/org.gradle.api.artifacts.dsl.DependencyAdder.html[`DependencyAdder`] page in the DSL reference.
 

--- a/subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc
+++ b/subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc
@@ -162,7 +162,7 @@ testing {
 }
 ```
 
-There are other related changes, such as:
+Other syntax effected by this change includes:
 
 - You cannot use `Provider<String>` as a dependency declaration.
 - You cannot use a `Map` as a dependency declaration for Kotlin or Java.

--- a/subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc
+++ b/subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc
@@ -163,9 +163,8 @@ testing {
 ```
 
 There may be other breaking changes related to dependency declarations in this block.
-For more information, see the updated <<jvm_test_suite_plugin.adoc#sec:declare_an_additional_test_suite, declare an additional test suite>> example in the JVM Test Suite Plugin section of the user guide, and the link:{groovyDslPath}/org.gradle.api.artifacts.dsl.DependencyAdder.html[`DependencyAdder`] page in the DSL reference.
 
-```kotlin
+For more information, see the updated <<jvm_test_suite_plugin.adoc#sec:declare_an_additional_test_suite, declare an additional test suite>> example in the JVM Test Suite Plugin section of the user guide, and the link:{groovyDslPath}/org.gradle.api.artifacts.dsl.DependencyAdder.html[`DependencyAdder`] page in the DSL reference.
 
 === Deprecations
 

